### PR TITLE
DO-15: Integrate member apis.

### PR DESCRIPTION
**Pull request by ayushmalik2110**
1) Refractored the code. 
2) Added schema form to the Member detail page. 
3) Integrated member apis to fetch all member and member by id. 
4) Moved config out side of the webpack build.

### CODE REVIEW

File: DO/src/components/higherOrderComponents/CSSVariableApplicator.jsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 
+//Todo this component can be modified more.
 class CSSVariableApplicator extends React.Component {
```
Comment (mohitsaladiya) [inline DO/src/components/higherOrderComponents/CSSVariableApplicator.jsx]: Please mention the purpose of the component in JSDOC

File: DO/src/services/categories/memberServices/memberSaga.js [hunk 2]
```diff
@@ -24,7 +24,9 @@ function* getMembersListSaga(action) {
   try {
     const state = yield select();
     const memberListAPIConfig = coreSelector.getAPIConfigByName(state, APIS.GET_MEMBERS_LIST);
-    const memberList  = yield getMembersList(memberListAPIConfig);
+    // todo get tenant id from the login details
+    const tenantId = 1;
```
Comment (mohitsaladiya) [inline DO/src/services/categories/memberServices/memberSaga.js]: Please fetch this tenantId dynamically

File: DO/src/utils/commonUtils/FormValidation.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import deepmerge from 'deepmerge';
+import { cloneDeep } from 'lodash';
+
+import customValidators from './customValidators';
+
+/**
+ * This method is responsible for read the custom validation on the form field in json
```
Comment (mohitsaladiya) [inline DO/src/utils/commonUtils/FormValidation.js]: Why aren't we using arrow functions in this file?

Comment (ayushmalik2110) [reply DO/src/utils/commonUtils/FormValidation.js]: Done

File: DO/src/components/categories/member/MemberDetails.jsx [hunk 4]
```diff
@@ -10,8 +10,16 @@ const {
...
+ * This component renders the member details along with its pictures.
+ */
 class MemberDetails extends React.Component {
 
+  /**
+   * This method render a JS library to display the member picture in album.
+   * @param memberDetails
+   * @returns {*}
+   */
   renderAlbum = (memberDetails) => {
```
[the remark this answers is not in the document]
Comment (ayushmalik2110) [reply DO/src/components/categories/member/MemberDetails.jsx]: Done.

File: DO/src/components/categories/member/MemberDetails.jsx [hunk 5]
```diff
@@ -32,8 +40,10 @@ class MemberDetails extends React.Component {
   renderInformation = (memberDetails) => {
```
[the remark this answers is not in the document]
Comment (ayushmalik2110) [reply DO/src/components/categories/member/MemberDetails.jsx]: Done.

File: DO/src/components/categories/member/MembersListView.jsx [hunk 6]
```diff
@@ -16,18 +18,23 @@ class MembersListView extends React.Component {
...
     };
-    this.rowRenderer = this.rowRenderer.bind(this);
   }
 
+  /**
+   * This method read the list view column header form config and render it.
+   * @param member
+   * @returns {*}
+   */
   renderMemberListViewHeader(member) {
```
[the remark this answers is not in the document]
Comment (ayushmalik2110) [reply DO/src/components/categories/member/MembersListView.jsx]: Done.

File: DO/src/components/categories/member/MembersListView.jsx [hunk 7]
```diff
@@ -39,7 +46,17 @@ class MembersListView extends React.Component {
...
+   * for each element of provided list and render rows.
+   * @param key
+   * @param index
+   * @param isScrolling
+   * @param isVisible
+   * @param style
+   * @returns {*}
+   */
+  rowRenderer =({ key, index, isScrolling, isVisible, style, }) => {
     const updatedStyle = {
```
[the remark this answers is not in the document]
Comment (ayushmalik2110) [reply DO/src/components/categories/member/MembersListView.jsx]: Done

File: DO/src/components/categories/member/MembersListView.jsx [hunk 8]
```diff
@@ -52,42 +69,75 @@ class MembersListView extends React.Component {
...
   }
 
+  /**
+   * This method trigger when user give some input into the search box.
+   * Ans it is also filter the data on the basis of search and set it to state.
+   * @param event
+   */
   search = (event) => {
     const searchKey = event.target.value;
     const options = {
```
[the remark this answers is not in the document]
Comment (ayushmalik2110) [reply DO/src/components/categories/member/MembersListView.jsx]: Done.

File: DO/src/components/higherOrderComponents/CSSVariableApplicator.jsx (hunk 1 above)
Comment (ayushmalik2110) [reply DO/src/components/higherOrderComponents/CSSVariableApplicator.jsx]: Done.

File: DO/src/services/categories/memberServices/memberSaga.js (hunk 2 above)
Comment (ayushmalik2110) [reply DO/src/services/categories/memberServices/memberSaga.js]: Done.

File: DO/src/utils/commonUtils/FormValidation.js (hunk 3 above)
Comment (ayushmalik2110) [reply DO/src/utils/commonUtils/FormValidation.js]: Done.

**Pull request merged**